### PR TITLE
Iiwa 203

### PR DESCRIPTION
**Pull request by mcdowellalex**
includes everything from IIWA 187

### CODE REVIEW

File: src/frontend-admin/src/components/pages/HardwareDetailPage/HardwareDetailEditPage.tsx [hunk 1]
```diff
@@ -0,0 +1,989 @@
...
+                        Ram: firstSectionData[3],
+                        LocalHHD: firstSectionData[4],
+                        NumberOfCores: firstSectionData[5],
+                        MFG: firstSectionData[6],
+                        SerialNumber: firstSectionData[7],
+                        IPAddress: firstSectionData[8],
+                        SAN: firstSectionData[9],
+                        Fqdn: firstSectionData[10],
+
+                        Virtualize: false, //NEED TO ADD THIS
```
Comment (kirbur) [inline src/frontend-admin/src/components/pages/HardwareDetailPage/HardwareDetailEditPage.tsx]: Is this added yet?

File: src/frontend-admin/src/components/pages/HardwareDetailPage/HardwareDetailPage.tsx [hunk 2]
```diff
@@ -1,29 +1,338 @@
...
+import {Group} from '../../reusables/Group/Group'
+import {HistoryLog} from '../../reusables/HistoryLog/HistoryLog'
 
 // Utils
+import {formatDate} from '../../../utilities/FormatDate'
 
 // Styles
-//import styles from './HardwareDetailPage.module.css'
+import styles from './HardwareDetailPage.module.css'
+import placeholder from '../../../content/Images/Placeholders/program-placeholder.png'
```
Comment (kirbur) [inline src/frontend-admin/src/components/pages/HardwareDetailPage/HardwareDetailPage.tsx]: wrong placeholder

File: src/frontend-admin/src/components/pages/HardwareDetailPage/HardwareDetailPage.tsx [hunk 3]
```diff
@@ -1,29 +1,338 @@
...
 }
 
-// Helpers
-
 // Primary Component
 export const HardwareDetailPage: React.SFC<IHardwareDetailPageProps> = props => {
     const {history, match} = props
 
+    const {
+        loginContextVariables: {accessToken, refreshToken /*, isAdmin*/},
```
Comment (kirbur) [inline src/frontend-admin/src/components/pages/HardwareDetailPage/HardwareDetailPage.tsx]: uncomment

File: src/frontend-admin/src/components/pages/HardwareDetailPage/HardwareDetailPage.tsx [hunk 4]
```diff
@@ -1,29 +1,338 @@
...
+    } = useContext(LoginContext)
+    const isAdmin = true //TODO: remove
```
Comment (kirbur) [inline src/frontend-admin/src/components/pages/HardwareDetailPage/HardwareDetailPage.tsx]: remove

File: src/frontend-admin/src/components/pages/HardwareDetailPage/HardwareDetailEditPage.tsx (hunk 1 above)
Comment (mcdowellalex) [reply src/frontend-admin/src/components/pages/HardwareDetailPage/HardwareDetailEditPage.tsx]: no

Comment (mcdowellalex) [reply src/frontend-admin/src/components/pages/HardwareDetailPage/HardwareDetailEditPage.tsx]: yes

File: src/frontend-admin/src/components/pages/DashboardPage/DashboardPage.tsx [hunk 5]
```diff
@@ -14,13 +14,14 @@ import {HorizontalBarChart} from '../../reusables/HorizontalBarChart/HorizontalB
...
 import {CostCard} from '../Dashboard/CostCard/CostCard'
+import {History} from 'history'
 
 // Context
 import {LoginContext} from '../../App/App'
 
 // Types
 interface IDashboardPageProps {
-    history: any
+    history: History
```
Comment (freakingawesome) [inline src/frontend-admin/src/components/pages/DashboardPage/DashboardPage.tsx]: Yay!

**Pull request merged**